tests/lib/thrift/ThriftTest/src/main.cpp

```diff
@@ -0,0 +1,147 @@
+  return std::shared_ptr<TServer>(
+      new TSimpleServer(processor, serverTransport, transportFactory, protocolFactory));
+}
+
+static void setup(void) {
+  static bool done = false;
+  if (done)
+    return;
+  done = true;
+
+  int rv;
+  pthread_attr_t attr;
+  pthread_attr_t* attrp = &attr;
+
+  if (IS_ENABLED(CONFIG_ARCH_POSIX)) {
+    attrp = NULL;
+  } else {
+    rv = pthread_attr_init(attrp);
+    zassert_equal(0, rv, "pthread_attr_init failed: %d", rv);
+    rv = pthread_attr_setstack(attrp, ThriftTest_server_stack, HELLO_SERVER_STACK_SIZE);
+    zassert_equal(0, rv, "pthread_attr_setstack failed: %d", rv);
+  }
+
+  // create the communication channel
+  rv = socketpair(AF_UNIX, SOCK_STREAM, 0, &context.fds.front());
+  zassert_equal(0, rv, "socketpair failed: %d", rv);
+
+  // set up server
+  context.server = setup_server();
+
+  // start the server
+  rv = pthread_create(&context.server_thread, attrp, server_func, nullptr);
+  zassert_equal(0, rv, "pthread_create failed: %d", rv);
+
+  // set up client
+  context.client = setup_client();
+}
+
+static void teardown(void) {}
+
+static void teardown_(void) {
+  void* unused;
+
+  context.server->stop();
+
+  pthread_join(context.server_thread, &unused);
+
+  for (auto& fd : context.fds) {
+    close(fd);
+  }
+}
+
+void test_main(void) {
+  ztest_test_suite(thrift_test, ztest_unit_test_setup_teardown(test_void, setup, teardown),
+                   ztest_unit_test_setup_teardown(test_string, setup, teardown),
+                   ztest_unit_test_setup_teardown(test_bool, setup, teardown),
+                   ztest_unit_test_setup_teardown(test_byte, setup, teardown),
+                   ztest_unit_test_setup_teardown(test_i32, setup, teardown),
+                   ztest_unit_test_setup_teardown(test_i64, setup, teardown),
+                   ztest_unit_test_setup_teardown(test_double, setup, teardown),
+                   ztest_unit_test_setup_teardown(test_binary, setup, teardown),
+                   ztest_unit_test_setup_teardown(test_struct, setup, teardown),
+                   ztest_unit_test_setup_teardown(test_nested_struct, setup, teardown),
+                   ztest_unit_test_setup_teardown(test_map, setup, teardown),
+                   ztest_unit_test_setup_teardown(test_string_map, setup, teardown),
+                   ztest_unit_test_setup_teardown(test_set, setup, teardown),
+                   ztest_unit_test_setup_teardown(test_list, setup, teardown),
+                   ztest_unit_test_setup_teardown(test_enum, setup, teardown),
+                   ztest_unit_test_setup_teardown(test_typedef, setup, teardown),
+                   ztest_unit_test_setup_teardown(test_nested_map, setup, teardown),
+                   ztest_unit_test_setup_teardown(test_exception, setup, teardown),
+                   ztest_unit_test_setup_teardown(test_multi_exception, setup, teardown));
+  ztest_run_test_suite(thrift_test);
+  teardown_();
```

Comment:
I just noticed this - it should be removed and teardown should be done in `teardown()`.